src/jit/decomposelongs.cpp

```diff
@@ -1581,9 +1581,17 @@ GenTree* DecomposeLongs::DecomposeMul(LIR::Use& use)
     Range().Remove(op1);
     Range().Remove(op2);
 
+#if defined(_TARGET_X86_)
     tree->gtOp.gtOp1 = op1->gtGetOp1();
     tree->gtOp.gtOp2 = op2->gtGetOp1();
     tree->SetOperRaw(GT_MUL_LONG);
+#elif defined(_TARGET_ARM_)
+    GenTree* mulTree = new (m_compiler, GT_MUL_LONG) GenTreeMulLong(TYP_LONG, op1->gtGetOp1(), op2->gtGetOp1());
+    mulTree->gtFlags = tree->gtFlags;
+    Range().InsertBefore(tree, mulTree);
+    use.ReplaceWith(m_compiler, mulTree);
+    Range().Remove(tree);
+#endif
 
```

Comment:
I think you can avoid this `#ifdef` entirely by changing `SetOperRaw` to `SetOper`. I'm not sure x86 depends on the `Raw` version, or that just happens to work.